torch_glow/src/PyTorchModelLoader.cpp

```diff
@@ -2373,15 +2373,39 @@ Error PyTorchModelLoader::loadLog(const torch::jit::Node *ptNode) {
 Error PyTorchModelLoader::loadMax(const torch::jit::Node *ptNode) {
   auto inputs = ptNode->inputs();
   auto outputs = ptNode->outputs();
-  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, 2, outputs, 1));
+  RETURN_IF_ERR(checkInputAndOutputSizes(inputs, -2, outputs, -1));
 
-  glow::NodeValue lhs;
-  ASSIGN_VALUE_OR_RETURN_ERR(lhs, getGlowNodeValueForValue(inputs[0]));
-  glow::NodeValue rhs;
-  ASSIGN_VALUE_OR_RETURN_ERR(rhs, getGlowNodeValueForValue(inputs[1]));
+  glow::NodeValue input;
+  ASSIGN_VALUE_OR_RETURN_ERR(input, getGlowNodeValueForValue(inputs[0]));
 
-  glow::MaxNode *glowNode = F_.createMax("max", lhs, rhs);
-  RETURN_ERR(addValueMapping(outputs[0], glowNode->getResult()));
+  if (inputs.size() == 2) {
+    glow::NodeValue rhs;
+    ASSIGN_VALUE_OR_RETURN_ERR(rhs, getGlowNodeValueForValue(inputs[1]));
+
+    glow::MaxNode *glowNode = F_.createMax("max", input, rhs);
+    RETURN_ERR(addValueMapping(outputs[0], glowNode->getResult()));
+  } else {
+    int64_t axis;
+    ASSIGN_VALUE_OR_RETURN_ERR(axis,
+                               iValToInt(getGlowIValueForValue(inputs[1])));
+    if (axis < 0)
+      axis = input.dims().size() + axis;
+
+    bool keepdim = false;
+    ASSIGN_VALUE_OR_RETURN_ERR(keepdim, iValToBool(getGlowIValueForValue(
+                                            ptNode->namedInput("keepdim"))));
+    glow::NodeValue maxNode = F_.createBatchedReduceMax(
+        "reduceMax", input, {static_cast<unsigned_t>(axis)});
```

Comment:
Could you please standardlize the node name? Like also use reduce_max here.

Reply:
Done.